app/models/foreman_kubevirt/kubevirt.rb

```diff
@@ -136,18 +140,16 @@ def create_vm(args = {})
       logger.debug("creating VM with the following options: #{options.inspect}")
 
       volumes = create_volumes_for_vm(options)
-
-      # FIXME: Add cloud-init support
-      # init = { 'userData' => "#!/bin/bash\necho \"fedora\" | passwd fedora --stdin"}
-
       interfaces, networks = create_network_devices_for_vm(options, volumes)
+      # Add clound init user data
+      user_data = options[:user_data].present? ? { "userData" => options[:user_data] } : nil
```

Comment:
user_data = { "userData" => options[:user_data] } if options[:user_data].present?
arguably looks better, but I can see how what you wrote might be clearer, so I will merge like this.

Reply:
in this context your proposal makes sense, I"ll change this.